radix-engine-tests/tests/system/threading.rs

```diff
@@ -0,0 +1,268 @@
+                    end_epoch_exclusive,
+                    min_proposer_timestamp_inclusive: None,
+                    max_proposer_timestamp_exclusive: None,
+                    intent_discriminator: 1,
+                })
+                .manifest_builder(|builder| {
+                    builder
+                        // EntireWorktop will ensure the buckets are passed as-is.
+                        .yield_to_parent((ManifestExpression::EntireWorktop,))
+                })
+                .sign(&sk1)
+                .build(),
+        )
+        .intent_header(IntentHeaderV2 {
+            network_id: NetworkDefinition::simulator().id,
+            start_epoch_inclusive,
+            end_epoch_exclusive,
+            min_proposer_timestamp_inclusive: None,
+            max_proposer_timestamp_exclusive: None,
+            intent_discriminator: 2,
+        })
+        .manifest_builder(|builder| {
+            builder
+                .lock_fee(account1, 3)
+                .call_method(component_address, "create_locked_bucket", (dec!(1),))
+                // EntireWorktop will ensure the buckets are passed as-is.
+                .yield_to_child("child", (ManifestExpression::EntireWorktop,))
+                // Free the bucket
+                .drop_all_proofs()
+                .try_deposit_entire_worktop_or_abort(account1, None)
+        })
+        .transaction_header(TransactionHeaderV2 {
+            notary_public_key: pk1.into(),
+            notary_is_signatory: false,
+            tip_basis_points: 0,
+        })
+        .sign(&sk1)
+        .notarize(&sk1)
+        .build();
+
+    let receipt = ledger.execute_transaction(&transaction, ExecutionConfig::for_test_transaction());
+    receipt.expect_commit_success();
+}
+
+// Arguably, we may disallow transferring references
+#[test]
+fn can_pass_global_and_direct_access_references() {
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+    let (pk1, sk1, account1) = ledger.new_allocated_account();
+    let (_, _, account) = ledger.new_allocated_account();
+    let vault = ledger.get_component_vaults(account, XRD).pop().unwrap();
+
+    let start_epoch_inclusive = ledger.get_current_epoch();
+    let end_epoch_exclusive = start_epoch_inclusive.after(1).unwrap();
+    let transaction = TransactionV2Builder::new()
+        .add_signed_child(
+            "child",
+            PartialTransactionV2Builder::new()
+                .intent_header(IntentHeaderV2 {
+                    network_id: NetworkDefinition::simulator().id,
+                    start_epoch_inclusive,
+                    end_epoch_exclusive,
+                    min_proposer_timestamp_inclusive: None,
+                    max_proposer_timestamp_exclusive: None,
+                    intent_discriminator: 1,
+                })
+                .manifest_builder(|builder| {
+                    builder
+                        // Unfortunately, there is no way to grab the received references
```

Comment:
This test doesn't show whether we can use those references? Can we try doing a vault freeze here perhaps?

Or wait I assume this just-works because it's already in the subintent manifest? So... passing references doesn't do anything really?